code/modules/ninja/martial_art.dm

```diff
@@ -0,0 +1,181 @@
+/*
+ *Contains:
+ * Creeping Widow martial art datum
+ * Creeping Widow MMB override datum
+ * grant_ninja_martial_art proc
+ * Creeping Widow injector
+ */
+
+
+// Creeping Widow injector - Single use nanomachine thing that teaches people the creeping widow style.
+
+
+/obj/item/weapon/creeping_widow_injector/
+	name = "strange injector"
+	desc = "A strange autoinjector made of a black metal.<br>You can see a green liquid through the glass."
+	icon = 'icons/obj/ninjaobjects.dmi'
+	icon_state = "injector"
+	attack_verb = list("poked", "prodded")
+	var/used = 0
+
+/obj/item/weapon/creeping_widow_injector/attack_self(mob/living/carbon/human/user as mob)
+	if(!used)
+		user.visible_message("<span class='warning'>You stick the [src]'s needle into your arm and press the button.", \
+			  "<span class='warning'>[user] sticks the [src]'s needle \his arm and presses the button.")
+		grant_ninja_martial_art(user)
+		user << "<span class='info'>The nanomachines in the [src] flow through your bloodstream. You have been taugh the ways of the <i>Creeping Widow</i>.<br>\
+				Your stikes on harm intent will deal more damage.<br>Using middle mouse button on a nearby person while on harm intent will send them flying backwards.<br>\
+				Your grabs will instantly be aggressive while you are using this style.<br>Using middle mouse button while on harm intent and behind a person will put them in a silencing choke hold.<br>\
+				Using middle mouse button on a nearby person while on disarm intent will wrench their wrist, causing them to drop what they are holding.</span>"
+		used = 1
+		icon_state = "injector-used"
+		desc = "A strange autoinjector made of a black metal.<br>It appears to be used up and empty."
+		return 0
+	else
+		user << "<span class='warning'>The [src] has been used already!</span>"
+		return 1
+
+// grant_ninja_martial_art proc - It is imperitive to ALWAYS use this when giving someone the Ninja martial art so that they have the middle mouse override as well.
+
+/proc/grant_ninja_martial_art(var/mob/living/carbon/human/target)
+	target.martial_art = new /datum/martial_art/ninja_martial_art()
```

Comment:
thiiss is really not how you are supposed to give people martial arts

The proper way to do it would be creating the datum and then proceeding to teach it to them as a permanent style, IE
```
var/datum/martial_art/ninja_martial_art/N = new
N.teach(target, 0)
```
If you do it this way, you can also override the `/datum/martial_art/ninja_martial_art/proc/teach` and create the `middleClickOverride` there.